pkg/cmd/secvulnOssindex.go

```diff
@@ -34,4 +44,238 @@ func init() {
 
 func secvulnOssindexExecute(cmd *cobra.Command, args []string) {
 	fmt.Printf("Galasa Build - Security Vulnerability OSS Index - version %v\n", rootCmd.Version)
+
+	// Find all target directories in the structure of the provided parent directory
+	findTargetDirectories(secvulnOssindexParentDir)
+
+	// Create the yaml report of all vulnerabilities found
+	createYamlReport()
+	fmt.Printf("Exported security vulnerability report to %s\n", secvulnOssindexOutput)
+
+}
+
+func findTargetDirectories(directory string) {
+
+	subDirs, err := ioutil.ReadDir(directory)
+	if err != nil {
+		fmt.Printf("Unable to read the sub-directories of directory %s, %v\n", directory, err)
+	}
+
+	for _, f := range subDirs {
```

Comment:
I would massively simplify this code by using path/filepath/Walk,  see :-
https://pkg.go.dev/path/filepath#Walk
Dont do it the same as it is in the example, ie embedded function,  use a standalone function as it is more readable.

Reply:
then in the function,   check if the file name is audit-report.json,
find the pom.xml from 2 directories above and then process.